org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/ModelBuilder.scala

```diff
@@ -0,0 +1,257 @@
+package org.scalaide.ui.internal.editor
+
+import scala.tools.nsc.Settings
+import scala.tools.nsc.interactive.CompilerControl
+import scala.tools.nsc.interactive.Global
+import scala.tools.nsc.reporters.ConsoleReporter
+import scala.reflect.internal.util.BatchSourceFile
+import scala.reflect.internal.util.SourceFile
+import scala.tools.nsc.interactive.Response
+import org.scalaide.core.compiler.IScalaPresentationCompiler
+import org.scalaide.logging.HasLogger
+import scala.collection.mutable.MutableList
+
+trait HasReturnType {
+  private var rt: Option[String] = None
+  def returnType = rt
+  def returnType_=(rt: Option[String]) = {
+    this.rt = rt
+  }
+}
+trait HasModifiers {
+  import scala.reflect.internal.Flags._
+  private[editor] var flags = 0L
+  def setFlags(f: Long) = {
+    flags = f
+  }
+  def isPrivate = (flags & PRIVATE) != 0
+  def isProtected = (flags & PROTECTED) != 0
+  def isTrait = (flags & TRAIT) != 0
+}
+abstract class NodeKey
+
+abstract class Node(val isLeaf: Boolean) extends HasModifiers {
+  def name: String
+  def parent: ContainerNode
+  var start: Int = 0
+  var end: Int = 0
+  def displayName: String
+  def key: NodeKey
+  def <= (that:Node):Boolean={
```

Comment:
Also, you shouldn't write such code. A comparison is supposed to be a side effect free operation. But yours is not.